controllers/authorino_controller.go

```diff
@@ -191,8 +192,15 @@ func (r *AuthorinoReconciler) buildAuthorinoDeployment(authorino *api.Authorino)
 	var containers []k8score.Container
 	var saName = authorino.Name + "-authorino"
 
-	if authorino.Spec.Image == "" {
-		authorino.Spec.Image = DefaultAuthorinoImage
+	image := authorino.Spec.Image
+
+	if image == "" {
+		image = env.GetString("RELATED_IMAGE_AUTHORINO", DefaultAuthorinoImage)
+	}
+
+	if image == "" {
+		// `DefaultAuthorinoImage can be empty string. But image cannot be or deployment will fail
+		image = "quay.io/kuadrant/authorino:latest"
```

Comment:
This is dangerous. `DefaultAuthorinoImage` should never be empty. If we're keeping this build-arg variable as an option to set the default Authorino image, then IMO the variable should be set at every build to a non-empty value.

If this variable can be empty, I think I'd rather break the operator than make it silently reset all Authorino instances in the cluster to `latest` if the `RELATED_IMAGE_AUTHORINO` is removed from the deployment.

Reply:
ok, panicking if it is empty